### PR TITLE
support more types

### PR DESCRIPTION
**Pull request by knassar702**
hello @Yush08 thanks for this crate, I added new types like integer, float, bool and I'm working on the array and hashmap types 

here's the new format
```
name:"string between \"\" "
bool:true
number:55
float:1.4
```

best regards

### CODE REVIEW

**Comment by Yush08** [comment]
Hello! Sorry for the late reply, I will definitely take a look at this PR and merge it soon!

Somehow the notification didn't pop up, I'll take a peek right now :P

File: src/lib.rs [hunk 1]
```diff
@@ -7,42 +7,66 @@ pub enum Error {
...
 }
 
 #[cfg(test)]
 mod tests {
     use crate::interpret;
+    use crate::Types;
 
     #[test]
     fn parsing_works() {
-        let config =
```
Comment (Yush08) [inline src/lib.rs:40]: I'd appreciate it if you could not remove these tests :)

File: src/lib.rs [hunk 2]
```diff
@@ -7,42 +7,66 @@ pub enum Error {
...
-            interpret("rust_is_awesome:true\ndeno_is_uncool:false\nname_of_cute_crab:Ferris!")
-                .unwrap();
-
-        assert_eq!(config.get("rust_is_awesome").unwrap(), &"true");
-        assert_eq!(config.get("deno_is_uncool").unwrap(), &"false");
-        assert_eq!(config.get("name_of_cute_crab").unwrap(), &"Ferris!");
+        let config = interpret("name:\"Rust\"\ndata:15\ntest:true").unwrap();
+        assert_eq!(config.get("name").unwrap(), &Types::String("Rust".to_string()));
+        assert_eq!(config.get("data").unwrap(), &Types::Integer(15));
+        assert_eq!(config.get("test").unwrap(), &Types::Boolean(true));
```
Comment (Yush08) [inline src/lib.rs:70]: Could you add a few more tests please? Especially with your object type.

**Review by Yush08 — CHANGES_REQUESTED** [review]
Thanks for the PR! Really appreciate it -- can you check on these requests please?

File: src/lib.rs (hunk 1 above)
Comment (knassar702) [reply src/lib.rs:40]: sure :+1:

**Review by gimbles — APPROVED** [review]

**Comment by gimbles** [comment]
Appreciate the help on this, I'll test it out locally and merge in a few minutes

**Pull request merged**